packages/fluentui/react-bindings/src/hooks/useClasses.ts

```diff
@@ -0,0 +1,218 @@
+    }
+  }
+
+  if (performanceFlags.enableBooleanVariablesCaching) {
+    if (_.isPlainObject(variables)) {
+      const hasOnlyBooleanVariables = Object.keys(variables).every(
+        variableName =>
+          variables[variableName] === null ||
+          typeof variables[variableName] === 'undefined' ||
+          typeof variables[variableName] === 'boolean',
+      );
+
+      if (!hasOnlyBooleanVariables) {
+        noVariableOverrides = false;
+      }
+    } else if (!_.isNil(variables)) {
+      noVariableOverrides = false;
+    }
+  }
+
+  const cacheEnabled = performanceFlags.enableStylesCaching && noInlineStylesOverrides && noVariableOverrides;
+
+  // Merge theme styles with inline overrides if any
+  let mergedStyles: ComponentSlotStylesPrepared;
+
+  if (noInlineStylesOverrides) {
+    mergedStyles = theme.componentStyles[displayName];
+  } else {
+    mergedStyles = mergeComponentStyles(
+      theme.componentStyles[displayName],
+      design && withDebugId({ root: design }, 'props.design'),
+      styles && withDebugId({ root: styles } as ComponentSlotStylesInput, 'props.styles'),
+    );
+  }
+
+  const styleParam: ComponentStyleFunctionParam = {
+    disableAnimations,
+    props: componentStylesProps,
+    theme,
+    rtl,
+    variables: resolvedVariables,
+  };
+
+  // Heads Up! Keep in sync with Design.tsx render logic
+  const rendererParam: RendererParam = {
+    direction: rtl ? 'rtl' : 'ltr',
+    disableAnimations,
+    displayName, // does not affect styles, only used by useEnhancedRenderer in docs
+    sanitizeCss: performanceFlags.enableSanitizeCssPlugin,
+  };
+
+  const propsCacheKey = cacheEnabled ? JSON.stringify(componentStylesProps) : '';
+  const variablesCacheKey =
+    cacheEnabled && performanceFlags.enableBooleanVariablesCaching ? JSON.stringify(variables) : '';
+  const componentCacheKey = cacheEnabled
+    ? `${displayName}:${propsCacheKey}:${variablesCacheKey}:${styleParam.rtl}${styleParam.disableAnimations}${resolvedVariablesHash}`
+    : '';
+
+  let resolvedStyles: ICSSInJSStyle;
+  const telemetryStylesStart = telemetry?.enabled ? performance.now() : 0;
+
+  if (cacheEnabled && noInlineStylesOverrides) {
+    // @ts-ignore
+    if (mergedStyles.root[componentCacheKey]) {
+      // @ts-ignore
+      resolvedStyles = mergedStyles.root[componentCacheKey];
+    } else {
+      resolvedStyles = mergedStyles.root(styleParam);
+      // @ts-ignore
+      mergedStyles.root[componentCacheKey] = resolvedStyles;
+    }
+  } else {
+    resolvedStyles = mergedStyles.root(styleParam);
+  }
+
+  if (telemetry?.enabled && telemetry.performance[displayName]) {
+    telemetry.performance[displayName].msResolveStylesTotal += performance.now() - telemetryStylesStart;
+  }
+
+  let classes: string = '';
+  const telemetryClassesStart = telemetry?.enabled ? performance.now() : 0;
+
+  if (cacheEnabled && noInlineStylesOverrides) {
+    // @ts-ignore
+    if (mergedStyles.root[componentCacheKey + 'classes']) {
```

Comment:
nit: can be moved to variable

```suggestion
    const cacheKey = `${componentCacheKey}classes` 
    if (mergedStyles.root[cacheKey ]) {
```